ra-optimisation/linear-range-action-rao/src/main/java/com/farao_community/farao/linear_range_action_rao/LinearRangeActionRao.java

```diff
@@ -35,7 +51,74 @@ public String getVersion() {
+                                                       ComputationManager computationManager, RaoParameters parameters,
+                                                       SensitivityComputationFactory sensitivityComputationFactory) {
+        SystematicSensitivityAnalysisResult sensiSaResults = SystematicSensitivityAnalysisService.runSensitivity(network, crac, computationManager, sensitivityComputationFactory);
+        if (sensiSaResults == null) {
+            LinearRangeActionRaoResult resultExtension = new LinearRangeActionRaoResult(LinearRangeActionRaoResult.SecurityStatus.UNSECURED);
+            RaoComputationResult raoComputationResult =  new RaoComputationResult(RaoComputationResult.Status.FAILURE);
+            raoComputationResult.addExtension(LinearRangeActionRaoResult.class, resultExtension);
+            return CompletableFuture.completedFuture(raoComputationResult);
+        }
+
+        LinearRangeActionRaoResult resultExtension = new LinearRangeActionRaoResult(LinearRangeActionRaoResult.SecurityStatus.SECURED);
+
+        // 1. do for pre
+        SensitivityComputationResults preSensi = sensiSaResults.getPrecontingencyResult();
+        List<MonitoredBranchResult> monitoredBranchResults = getMonitoredBranchResultList(crac, preSensi, resultExtension);
+        PreContingencyResult preRao = new PreContingencyResult(monitoredBranchResults);
+
+        // 2. do for each contingency
+        Map<Contingency, SensitivityComputationResults>  mapSensi = sensiSaResults.getResultMap();
+        List<ContingencyResult> contingencyResultsRao = new ArrayList<>();
+        for (Contingency contingency : mapSensi.keySet()) {
+            String idContSensi = contingency.getId();
+            String nameContSensi = contingency.getName();
+
+            SensitivityComputationResults sensitivityComputationResults = mapSensi.get(contingency);
+            List<MonitoredBranchResult> tmpMonitoredBranchResults = getMonitoredBranchResultList(crac, sensitivityComputationResults, resultExtension);
+            ContingencyResult contingencyResult = new ContingencyResult(idContSensi, nameContSensi, tmpMonitoredBranchResults);
+
+            contingencyResultsRao.add(contingencyResult);
+        }
+
+        RaoComputationResult raoComputationResult = new RaoComputationResult(RaoComputationResult.Status.SUCCESS, preRao, contingencyResultsRao);
+        raoComputationResult.addExtension(LinearRangeActionRaoResult.class, resultExtension);
+
+        // 4. return
+        return CompletableFuture.completedFuture(raoComputationResult);
+    }
+
+    private List<MonitoredBranchResult> getMonitoredBranchResultList(Crac crac, SensitivityComputationResults preSensi, LinearRangeActionRaoResult resultExtension) {
+        List<MonitoredBranchResult> returnlist = new ArrayList<>();
+        crac.getCnecs().forEach(cnec -> {
+            LOGGER.info("Cnec: " + cnec.getId());
+            preSensi.getSensitivityValues().forEach(sensitivityValue -> {
```

Comment:
Not OK: with this double loop (on CNECs and on sensitivityValues), and for a test with 10 CNECs, this function returns me 100 MonitoredBranchResult (instead of 10).

I guess there is a filter missing.

Reply:
fixed